### PR TITLE
Composer CICD pipeline tweaks plus

### PR DESCRIPTION
**Pull request by BruceHaley**
This PR is identical to and supersedes PR #8 with one additional change:

The file BasicAssistant/BasicAssistant/Nuget.config has been deleted to fix the build check error:

`
NuGet Security Analysis found 1 package manifests with myget feed declared. MyGet feeds are not allowed by Microsoft policy.
`
## Details (same as #8)
- Removed 3 parameters that are never used.
- Let luisEndpoint URL be set from an ADO variable: Composer allows selecting any of 3 regions. This change supports cases where regions other than "West US" have been selected. It also supports endpoints generated from the LUIS portal and imported into Composer. Those URL names are not related to region.
- Change Zip file reference from "${{ parameters.webAppName }}.zip" to "\*.zip": The generated .zip file name actually matches the project name, not the web app name. There was no var carrying the project name, so I opted for "\*". This works because there is only 1 zip file found in the zipDeploy folder.
- Conditionals added for running LUIS and Q&A publishing tasks: This prevents the pipeline from breaking when the bot has no Q&A or no LUIS component.

## Tests
Test pipelines can be found here:
https://fuselabs.visualstudio.com/SDK_v4/_build/index?definitionId=1463
https://dev.azure.com/FuseLabs/SDK_v4/_build/results?buildId=297624

### CODE REVIEW

File: BasicAssistant/BasicAssistant/Nuget.config [hunk 1]
```diff
@@ -1,8 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
```
Comment (gabog) [inline BasicAssistant/BasicAssistant/Nuget.config:1]: I don't know if we should remove this from the sample. Microsoft policies prevent us from using multiple nuget feeds but that's not the case for our customers.
Also, this file is generated by composer templates, the goal of this sample is to deploy composer created bots so I wouldn't want to alter the code generated by composer

File: build/yaml/buildAndDeploy.yaml [hunk 2]
```diff
@@ -57,23 +57,7 @@ parameters:
   default: $(BOTPROJECTNAME) 
 
 # LUIS Parameters
-# General
-- name: luisAppId
-  displayName: LUIS app ID
-  type: string
-  default: $(LUISAPPID)
-  
 # LUIS Authoring (used to build, train and publish LUIS models)
```
Comment (gabog) [inline build/yaml/buildAndDeploy.yaml:60]: I don't think we should remove the `luisAuthoringRegion` parameter, if anything we should probably update Build-LUIS.ps1 to take it and use it. 
Right now we are not passing it in and the call to `bf:luis:build` in Build-LUIS.ps1 doesn't pass it which means that is defaulting to westus and it won't support other authoring regions.

image

File: BasicAssistant/BasicAssistant/Nuget.config (hunk 1 above)
Comment (BruceHaley) [reply BasicAssistant/BasicAssistant/Nuget.config:1]: FYI this is my PR removing nuget.config from the Composer bots: https://github.com/microsoft/BotFramework-Composer/pull/9094 
Nuget.config is a legacy from before the SDK went public. It is no longer useful. I.e. dead code.

**Review by gabog — CHANGES_REQUESTED** [review]
See the comments and ping me if you have any quesitons

File: BasicAssistant/BasicAssistant/Nuget.config (hunk 1 above)
Comment (gabog) [reply BasicAssistant/BasicAssistant/Nuget.config:1]: OK, is this file also present in the composer template components? (empty bot, core bot, etc.)

Can you check with Ryan if this is the case and it needs to be removed from the other packages?

image

Comment (BruceHaley) [reply BasicAssistant/BasicAssistant/Nuget.config:1]: It is not in any of the templates. I researched and tested this. It is injected by a piece of legacy code in Composer when a bot project is created.

File: build/yaml/buildAndDeploy.yaml (hunk 2 above)
Comment (BruceHaley) [reply build/yaml/buildAndDeploy.yaml:60]: Done: `luisAuthoringRegion` added back as a parameter, and plumbed through Build-LUIS.ps1.

**Review by gabog — APPROVED** [review]

**Pull request merged**